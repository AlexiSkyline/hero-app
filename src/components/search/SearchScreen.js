import React from 'react';
import { useForm } from '../../hooks/useForm';
import { heroes } from '../data/heroes';
import { HeroCard } from '../heroes/HeroCard';

export const SearchScreen = () => {
    const heroesFilteres = heroes;
    
    const [ formValues, handleInputChange ] = useForm({ searchText: '' });
    const { searchText } = formValues;

    const handleSearch = (e) => {
        e.preventDefault();
        console.log( searchText );
    }

    return (
        <div>
            <h1>Search Screen</h1>
            <hr />

            <div className='row'>
                <div className='col-5'>
                    <h4>Search Form</h4>
                    <hr />

                    <form
                        onSubmit={ handleSearch }
                    >
                        <input 
                            type='text'
                            placeholder='Find your Hero'
                            className='form-control'
                            name='searchText'
                            value={ searchText }
                            onChange={ handleInputChange }
                        />

                        <button
                            type='submit'
                            className='btn m-1 btn-block btn-outline-primary'
                        >
                            Search...
                        </button>
                    </form>
                </div>

                <div className='col-7'>
                    <h4>Results </h4>
                    <hr />

                    {
                        heroesFilteres.map( hero => (
                            <HeroCard 
                                key={ hero.id }
                                hero={ hero }
                            />
                        ))
                    }
                </div>
            </div>
        </div>
    );
}