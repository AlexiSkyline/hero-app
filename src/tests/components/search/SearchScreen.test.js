import { mount } from 'enzyme';
import { MemoryRouter, Route } from 'react-router-dom';
import { SearchScreen } from '../../../components/search/SearchScreen';

describe( 'Test in <SearchScreen />', () => {
    test( 'Should displayed correctly with the  default value', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/search' ]}>
                <Route path='/search' component={ SearchScreen }/>
            </MemoryRouter>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.alert-info').text().trim() ).toBe('Search a hero');
    });

    test( 'Should display Batman and the input with the value of the queryString', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/search?q=batman'] }>
                <Route path='/search' component={ SearchScreen }/>
            </MemoryRouter>
        );

        expect( wrapper.find('input').prop('value') ).toBe('batman');
        expect( wrapper ).toMatchSnapshot();
    });

    test( 'Should display Error if no hero found', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/search?q=IronSpider']}>
                <Route path='/search' component={ SearchScreen }/>
            </MemoryRouter>
        );

        expect( wrapper.find('.alert-danger').text().trim() ).toBe('There is no a hero with IronSpider');
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('Should call the push for the history', () => {
        const historyMock = {
            push: jest.fn(),
        }

        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/search?q=batman' ]}>
                <Route 
                    path='/search'
                    component={ () => <SearchScreen history={ historyMock }/>}
                />
            </MemoryRouter>
        );

        wrapper.find('input').simulate('change', {
            target: {
                name: 'searchText',
                value: 'batman'
            }
        });

        wrapper.find('form').prop('onSubmit')({
            preventDefault(){}
        });

        expect( historyMock.push ).toHaveBeenCalledWith('?q=batman');
    });
    
});