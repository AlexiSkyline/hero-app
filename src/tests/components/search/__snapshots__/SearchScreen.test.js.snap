// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SearchScreen /> Should display Batman and the input with the value of the queryString 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your Hero"
          type="text"
          value="batman"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results 
      </h4>
      <hr />
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Batman"
              className="card-img"
              src="./assets/heroes/dc-batman.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Batman
              </h5>
              <p
                className="card-text"
              >
                Bruce Wayne
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Detective Comics #27
                </small>
              </p>
              <a
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                MÃ¡s...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test in <SearchScreen /> Should display Error if no hero found 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your Hero"
          type="text"
          value="IronSpider"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results 
      </h4>
      <hr />
      <div
        className="alert alert-danger"
      >
        There is no a hero with 
        IronSpider
      </div>
    </div>
  </div>
</div>
`;

exports[`Test in <SearchScreen /> Should displayed correctly with the  default value 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your Hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results 
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
